tests/api/test_post_api.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+import json
+
+from tests.setup_database import Setup
+from tests.utils import OpenEventTestCase
+from tests.auth_helper import login, register
+from tests.api.utils import create_event, get_path
+from tests.api.utils_post_data import *
+
+from open_event import current_app as app
+
+
+class TestPostApi(OpenEventTestCase):
+    """
+    Test POST APIs against 401 (unauthorized) and
+    200 (successful) status codes
+    """
+    def setUp(self):
+        self.app = Setup.create_app()
+        with app.test_request_context():
+            create_event()
+
+    def login_user(self):
+        """
+        register a user and login
+        """
+        register(self.app, 'test', 'test@gmail.com', 'test')
+        login(self.app, 'test', 'test')
+
+    def _test_model(self, name, data):
+        """
+        Tests -
+        1. Without login, try to do a POST request and catch 401 error
+        2. Login and match 200 response code and correct response data
+        """
+        path = get_path(1, name + 's')
+        response = self.app.post(
+            path,
+            data=json.dumps(data),
+            headers={'content-type': 'application/json'}
+        )
+        self.assertEqual(401, response.status_code, msg=response.data)
+        # login and send the request again
+        self.login_user()
+        response = self.app.post(
+            path,
+            data=json.dumps(data),
+            headers={'content-type': 'application/json'}
+        )
+        self.assertEqual(200, response.status_code, msg=response.data)
+        self.assertIn('Test' + str(name).title(), msg=response.data)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [No value for argument 'container' in method call](https://www.codacy.com/app/triplez/open-event-orga-server/file/2653967465/issues/source?bid=3233864&fileBranchId=3324785#l51)